examples/positron-django/src/positron/app.py

```diff
@@ -0,0 +1,61 @@
+import os
+import sys
+import socketserver
+from threading import Thread, Event
+from wsgiref.simple_server import WSGIServer
+
+import django
+from django.core.servers.basehttp import WSGIRequestHandler
+from django.core.handlers.wsgi import WSGIHandler
+
+import toga
+
+
+class ThreadedWSGIServer(socketserver.ThreadingMixIn, WSGIServer):
+    pass
+
+
+class Positron(toga.App):
+    def web_server(self):
+        # Use port 0 to let the server select an available port.
+        self._httpd = ThreadedWSGIServer(("127.0.0.1", 0), WSGIRequestHandler)
+        self._httpd.daemon_threads = True
+
+        os.environ["DJANGO_SETTINGS_MODULE"] = "webapp.settings"
+        django.setup(set_prefix=False)
+        wsgi_handler = WSGIHandler()
+        self._httpd.set_app(wsgi_handler)
+
+        self.is_serving.set()
+        self._httpd.serve_forever()
```

Comment:
These lines should be swapped, otherwise the WebView might try to connect to the server before it starts serving.

Reply:
You'd think so, but:

1. `serve_forever()` is the blocking call, so the event can't be triggered second.
2. `socket.getsockname()` is *also* a blocking call until the socket is able to accept a connection.

So - in practice, `is_serving` is really only confirming that `self._httpd` exists; once it does, you can call `getsockname()` and it won't proceed until there's an active socket.

I've renamed the event, and added some other clarifications and docs to reflect this.